### PR TITLE
[quantization/optimization] Sink Rescale node down past certain nodes.

### PR DESCRIPTION
**Pull request by rdzhabarov**
Validated on fr2en as well.

### CODE REVIEW

File: lib/Optimizer/GraphOptimizer.cpp [hunk 1]
```diff
@@ -951,6 +951,85 @@ static void optimizeQuantization(Function *F) {
   optimizeQuantizedMaxSplat(F);
 }
 
+/// Sink Rescale nodes down when possible.
+static bool sinkRescaleQuantizedNode(Function *F) {
+  bool changed = false;
+  for (auto const &node : F->getNodes()) {
+    // Sink Rescale below Reshape node.
```
Comment (nadavrot) [inline lib/Optimizer/GraphOptimizer.cpp]: Can you add a comment with the formula that explains what's the transformation:
// reshape(rescale(X))  -> rescale(reshape(X))

File: tests/unittests/graphOptzTest.cpp [hunk 2]
```diff
@@ -665,3 +665,23 @@ TEST_F(GraphOptz, FuseRescaleIntoArithmetic) {
   EXPECT_EQ(mul->getNthInput(0).getType(), rescaleOutTy);
   EXPECT_EQ(div->getNthInput(0).getType(), rescaleOutTy);
 }
+
+TEST_F(GraphOptz, sinkRescaledQuantizedNode) {
+  // Eliminate rescales.
```
Comment (nadavrot) [inline tests/unittests/graphOptzTest.cpp]: // Check that we eliminate rescale nodes by sinking them into other operators.

File: lib/Optimizer/GraphOptimizer.cpp (hunk 1 above)
Comment (rdzhabarov) [reply lib/Optimizer/GraphOptimizer.cpp]: fixed.

File: tests/unittests/graphOptzTest.cpp (hunk 2 above)
Comment (rdzhabarov) [reply tests/unittests/graphOptzTest.cpp]: fixed, thanks.

**Pull request merged**